tests/integration_tests/functional/test_uffd.py

```diff
@@ -100,3 +105,70 @@ def test_unbinded_socket(bin_cloner_path, test_microvm_with_api):
     assert "Load microVM snapshot error: Cannot connect to UDS in order to" \
            " send information on handling guest memory page-faults due to: " \
            "Connection refused (os error 111)" in response.text
+
+
+def test_valid_handler(bin_cloner_path,
+                       test_microvm_with_api,
+                       uffd_handler_paths):
+    """
+    Test valid uffd handler scenario.
+
+    @type: functional
+    """
+    logger = logging.getLogger("uffd_unbinded_socket")
+
+    logger.info("Create snapshot")
+    snapshot = create_snapshot(bin_cloner_path)
+
+    logger.info("Load snapshot, mem %s", snapshot.mem)
+    vm_builder = MicrovmBuilder(bin_cloner_path)
+    vm = test_microvm_with_api
+    vm.spawn()
+
+    jailed_mem = vm.create_jailed_resource(snapshot.mem)
+
+    # Copy page fault binary into chroot of microVM.
+    jailed_handler = os.path.join(vm.chroot(), "valid_handler")
+    shutil.copyfile(uffd_handler_paths['valid_handler'], jailed_handler)
+    # Change its permissions to the UID and GID of the jail.
+    cmd = 'chown {}:{} {}'.format(
+        vm.jailer.uid, vm.jailer.gid, jailed_handler)
+    run_cmd(cmd)
+
+    real_root = os.open("/", os.O_RDONLY)
+
+    os.chroot(vm.chroot())
+    os.chdir('/')
+    st = os.stat("valid_handler")
+    os.chmod("valid_handler", st.st_mode | stat.S_IEXEC)
+    socket_path = "/firecracker-uffd.sock"
+
+    proc = Popen(
+        ["/valid_handler", socket_path, jailed_mem],
+        stdout=PIPE,
+        stderr=PIPE,
+        bufsize=1)
+    try:
+        outs, errs = proc.communicate(timeout=1)
+        print(outs)
+        print(errs)
+        assert False, "Could not start PF handler!"
+    except TimeoutExpired:
+        print("This is the good case!")
```

Comment:
I'd like to have a helper function doing this setup and launching the handler process, including chroot-ing in and out of the vm jailed root

Reply:
Done